x/ccv/provider/keeper/misbehaviour.go

```diff
@@ -0,0 +1,233 @@
+	}
+
+	logger.Info(
+		"confirmed equivocation light client attack",
+		"byzantine validators slashed, jailed and tombstoned", provAddrs,
+	)
+
+	return nil
+}
+
+// GetByzantineValidators returns the validators that signed both headers.
+// If the misbehavior is an equivocation light client attack, then these
+// validators are the Byzantine validators.
+func (k Keeper) GetByzantineValidators(ctx sdk.Context, misbehaviour ibctmtypes.Misbehaviour) (validators []*tmtypes.Validator, err error) {
+	// construct the trusted and conflicted light blocks
+	lightBlock1, err := headerToLightBlock(*misbehaviour.Header1)
+	if err != nil {
+		return validators, err
+	}
+	lightBlock2, err := headerToLightBlock(*misbehaviour.Header2)
+	if err != nil {
+		return validators, err
+	}
+
+	// Check if the misbehaviour corresponds to an Amnesia attack,
+	// meaning that the conflicting headers have both valid state transitions
+	// and different commit rounds. In this case, we return no validators as
+	// we can't identify the byzantine validators.
+	//
+	// Note that we cannot differentiate which of the headers is trusted or malicious,
+	if !headersStateTransitionsAreConflicting(*lightBlock1.Header, *lightBlock2.Header) && lightBlock1.Commit.Round != lightBlock2.Commit.Round {
+		return validators, nil
+	}
+
+	// compare the signatures of the headers
+	// and return the intersection of validators who signed both
+
+	// create a map with the validators' address that signed header1
+	header1Signers := map[string]int{}
+	for idx, sign := range lightBlock1.Commit.Signatures {
+		if sign.Absent() {
+			continue
+		}
+		header1Signers[sign.ValidatorAddress.String()] = idx
+	}
+
+	// iterate over the header2 signers and check if they signed header1
+	for sigIdxHeader2, sign := range lightBlock2.Commit.Signatures {
+		if sign.Absent() {
+			continue
+		}
+		if sigIdxHeader1, ok := header1Signers[sign.ValidatorAddress.String()]; ok {
+			if err := verifyLightBlockCommitSig(*lightBlock1, sigIdxHeader1); err != nil {
+				return nil, err
+			}
+
+			if err := verifyLightBlockCommitSig(*lightBlock2, sigIdxHeader2); err != nil {
+				return nil, err
+			}
+
+			_, val := lightBlock1.ValidatorSet.GetByAddress(sign.ValidatorAddress)
+			validators = append(validators, val)
+		}
+	}
+
+	return validators, nil
+}
+
+// headerToLightBlock returns a CometBFT light block from the given IBC header
+func headerToLightBlock(h ibctmtypes.Header) (*tmtypes.LightBlock, error) {
+	sh, err := tmtypes.SignedHeaderFromProto(h.SignedHeader)
+	if err != nil {
+		return nil, err
+	}
+
+	vs, err := tmtypes.ValidatorSetFromProto(h.ValidatorSet)
+	if err != nil {
+		return nil, err
+	}
+
+	return &tmtypes.LightBlock{
+		SignedHeader: sh,
+		ValidatorSet: vs,
+	}, nil
+}
+
+// CheckMisbehaviour checks that headers in the given misbehaviour forms
+// a valid light client attack from an ICS consumer chain and that the light client isn't expired
```

Comment:
> [...] and that the light client isn't expired

Where's this check performed?
I think we can remove this expiration message from the comment. Additionally, maybe we can say "[...] a valid light client attack on a light client that tracks an ICS consumer chain" instead of "[...]  a valid light client attack from an ICS consumer chain"